server/graphql/v2/object/Order.js

```diff
@@ -201,6 +202,17 @@ export const Order = new GraphQLObjectType({
           return order; // Individual fields are set by OrderPermissions resolvers
         },
       },
+      data: {
+        type: GraphQLJSON,
+        description: 'Data related to the order',
+        async resolve(order) {
+          return {
+            customData: {
+              pledgeCurrency: order?.data?.customData?.pledgeCurrency,
+            },
+          };
+        },
+      },
```

Comment:
Can you propose something that is less **ad-hoc** and a bit more future proof? See for example: https://github.com/opencollective/opencollective-api/blob/main/server/graphql/v2/object/PaymentMethod.js#L87

```
// Protect and limit fields
const allowedFields = [];
return pick(order.data, allowedFields);
```

@Betree any other example of data field sanitization?

Reply:
@znarf the example you gave is good, another one is the same one on GQLV1 https://github.com/opencollective/opencollective-api/blob/84b4118c6c63fdab8631b385bce533e94c330d88/server/graphql/v1/types.js#L1763.

Note that pick supports sub-fields, so we can do:

```suggestion
      data: {
        type: GraphQLJSON,
        description: 'Data related to the order',
        async resolve(order) {
          return pick(order.data, ['customData.pledgeCurrency']);
        },
      },
```

Reply:
Btw why storing this in `customData`? This field is normally supposed to store values generated from tier's `customFields`, as per https://github.com/opencollective/opencollective-api/blob/84b4118c6c63fdab8631b385bce533e94c330d88/server/graphql/v2/input/OrderCreateInput.ts#L69.

If we want to store additional values on order, we can just store them directly in `order.data`.

Reply:
I've corrected the `data` field sanitation by using the `pick` method as suggested. 

As for the user of `order.data` instead of `customData` I agree and if no objections will do this in a separate PR at this might need changes both front and api. 🤔

Reply:
Issue opened at, https://github.com/opencollective/opencollective/issues/4607